openstack/clustering/v1/webhooks/results.go

```diff
@@ -0,0 +1,31 @@
+package webhooks
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// GetResult is the response of a Get operations. Call its Extract method to
+// interpret it as a Version.
+type GetResult struct {
+	commonResult
+}
+
+type TriggerResult struct {
+	commonResult
+}
+
+// Webhook represents a detailed webhook
+type Webhook struct {
+	Action string `json:"action"`
+}
```

Comment:
The result of the Trigger action is defined here:

https://github.com/openstack/senlin/blob/c0ef921768046a8795aa1f231ec6fafd5e67f3ea/senlin/engine/service.py#L2518

```
return {'action': action_id}
```

Since "webhook" isn't in the above body, I don't think we should call this struct `Webhook`. Since the result of this action is only returning one field, we should be able to get away with removing this struct entirely and changing the `Extract` method to look like:

```go
func (r commonResult) Extract() (string, err) {
  var s struct {
    Action string `json:"action"`
  }
  err := r.ExtractInto(&s)
  return s.Action, err
}
```

Reply:
Yes, your suggestion is better and cleaner